components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -271,11 +272,12 @@ private static String getMultiAttributeSeparator(String username,
         return claimSeparator;
     }
 
-    private static Map<String, String> getUserClaimsFromUserStore(String username,
+    private static Map<String, String> getUserClaimsFromUserStore(String username, String userStoreDomain,
                                                                   UserRealm realm,
                                                                   List<String> claimURIList) throws UserStoreException {
 
-        UserStoreManager userstore = realm.getUserStoreManager();
+        //Retrieve the UserStoreManager for the authenticated user's user-store domain
+        UserStoreManager userstore = realm.getUserStoreManager().getSecondaryUserStoreManager(userStoreDomain);
```

Comment:
Can userstore be null?

Reply:
No, it will be primary. Or else, an exception will be thrown. Null is not returned.

Reply:
unlikely, user store can be null. But added a null check